tests/fsharpqa/Source/Conformance/ObjectOrientedTypeDefinitions/DelegateTypes/E_InvalidDelegateDefinition.fs

```diff
@@ -0,0 +1,11 @@
+// #Regression #Conformance #ObjectOrientedTypes #Delegates 
```

Comment:
We try not to add any new tests to the `fsharpqa`, I've mentioned about the ComponentTests in the https://github.com/dotnet/fsharp/issues/8299#issuecomment-1115849356, sorry for the delay.

Reply:
No worries :) . Thanks for taking the time . I will update this PR according with your suggestion

Reply:
@vzarytovskii I realised that the #8299 talks about to **show a syntax error**  for this rather that a compilation error. 

I’m a little confused the test that I added in this PR they succeeded I was expecting them to fail As if you send interactive it show an error 😢.

Reply:
Yeah, when you call `compile` it will use FCS (i.e. compiler), and not fsi/interactive session.
Do you want to test compiler or fsi?

Reply:
> Yeah, when you call `compile` it will use FCS (i.e. compiler), and not fsi/interactive session.
> Did you intend to test compiler or fsi?


I think the expectation would to have one for **compiler**(that show a syntax error) and one for **fsi/interactive** session(that shows an error) . So they are both consistent .

Reply:
Yeah, so for compiler, you can simply leave this this test, and then fix the compiler itself, so the test passes (i.e. the compiler expected to fail).
For the FSI, you can use the `runFsi` function, here's the example of such test:
https://github.com/dotnet/fsharp/blob/d307f8d5091d9d1ac6e7e66948d4aac1596ab654/tests/FSharp.Compiler.ComponentTests/Scripting/Interactive.fs#L18-L29

Feel free to ping me on discord/slack if you have any questions.